src/php_spx.c

```diff
@@ -924,6 +924,10 @@ static void http_ui_handler_shutdown(void)
         ui_uri = "/index.html";
     }
 
+    if (ui_uri[0] != '/' || strstr(ui_uri, "/../") != NULL) {
```

Comment:
Perhaps realpath() would be useful to get rid of all relative paths, and then checking for the base directory path? Not an expert.

Reply:
Yes, I wasn't aware that realpath is part of stdlib.h, this is indeed a better solution. My first patch was clearly there to bring a fix ASAP.